_posts/2019/2019-01-13-use-multiple-migrations-when-adding-database-constraints.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Use Multiple Migrations When Adding Database Constraints
+description: You need to avoid a long table rebuild and a write-lock
+layout: article
+category: ruby
+image:
+  base: '2018/use-multiple-migrations-when-adding-database-constraints'
+  alt: 'Locks'
+  source: 'https://unsplash.com/photos/8_NI1WTqCGY'
+  credit: 'Marcos Mayer'
+---
+
+Adding constraints to your application at the database layer is a good idea as it provides an extra layer of quality control on top of the data powering your application.
+
+One way to do that is to add default values or constraints (like making sure a field can’t be blank), this is easy to do with Rails’ migrations. But beware...
+
+
+## Instead of…
+
+…adding a column and stipulating a default non-null value all in one go.
+
+```ruby
+class AddComposerToSongs < ActiveRecord::Migration[5.2]
+  def change
+    add_column :songs, :composer, :string, default: "Lin-Manuel Miranda", null: false
+  end
+end
+```
+
+
+## Use…
+
+…a multiple migration strategy to add a field with a constraint to your databases
```

Comment:
* a 'multiple migration' strategy ?  to emphasise that the strategy can be referred to as 'multiple migration'.

Reply:
And add a full-stop after 'databases'.